### PR TITLE
Shape command, with flag for size (closes #22, closes #21), see also #29

### LINKED ISSUE #21: `shape` command backend
A command which takes a dataset path, and returns a tuple of the shape of that dataset

### LINKED ISSUE #22: `shape` command cli
Frontend to #21

### PR DESCRIPTION
**Pull request by ksunden**

### CODE REVIEW

**Comment by ksunden** [comment]
One comment, the current implementation has an exception if the referenced item is not a dataset.

We could make it print the shape for each dataset in a group (possibly with a `-r` flag as an option)

File: src/h5cli/cli.py [hunk 1]
```diff
@@ -104,6 +104,23 @@ def do_bang(self, args, opts=None):
...
+        make_option('-n', '--number',action='store_true', help='print number of elements, rather than full shape'),
+        make_option('-m', '--maxshape',action='store_true', help='print maximum shape, rather than current shape')
+    ])
+    def do_shape(self, args, opts=None):
+        """Print the shape of a dataset."""
+        if opts.number:
+            print(self.explorer[args[0]].size)
+            return False
+
+        if opts.maxshape:
```
Comment (lphk92) [inline src/h5cli/cli.py]: Make this an `elif`

File: src/h5cli/cli.py [hunk 2]
```diff
@@ -104,6 +104,23 @@ def do_bang(self, args, opts=None):
...
+            print(self.explorer[args[0]].maxshape)
+            return False
+
+        print(self.explorer[args[0]].shape)
```
Comment (lphk92) [inline src/h5cli/cli.py]: Put this in an else, and just return False once at the bottom of the function

File: src/h5cli/cli.py [hunk 3]
```diff
@@ -104,6 +104,23 @@ def do_bang(self, args, opts=None):
     def do_clear(self, args):
         os.system('clear')
 
+    @options([
+        make_option('-n', '--number',action='store_true', help='print number of elements, rather than full shape'),
+        make_option('-m', '--maxshape',action='store_true', help='print maximum shape, rather than current shape')
```
Comment (lphk92) [inline src/h5cli/cli.py]: From the logic in the function it seems like these two options are mutually exclusive. Is there a way to specify that these options can't go together?

**Review by lphk92 — CHANGES_REQUESTED** [review]
Left some comments on structure of command. In summary:

- Make it an actually set if..elif..else block in the function and just return once
- We need to have some way of telling the user that the two options are mutually exclusive.

**Comment by ksunden** [comment]
Not sure we can make them strictly mutually exclusive, but we can make them write to the same destination variable, in which opt parse will result in the value from the last option

**Comment by ksunden** [comment]
```
[ultrafast.hdf5: /] > shape 1D/T
(81,)
[ultrafast.hdf5: /] > shape -n 1D/T
81
[ultrafast.hdf5: /] > shape -nm 1D/T
(81,)
[ultrafast.hdf5: /] > shape -nmn 1D/T
81
```

**Comment by ksunden** [comment]
ping @lphk92

**Review by lphk92 — APPROVED** [review]

**Pull request merged**